dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-reports-reconciliation-summary.controller/helpers.ts

```diff
@@ -0,0 +1,167 @@
+  getReportPeriodStartForSubmissionMonth,
+  getReportPeriodStartForUtilisationReport,
+  getSubmissionMonthForReportPeriodStart,
+  isEqualReportPeriodStart,
+} from '../../../../utils/report-period';
+
+type UtilisationReportForSubmissionMonth = {
+  submissionMonth: IsoMonthStamp;
+  report: UtilisationReport | null;
+};
+
+type SummaryItemForSubmissionMonth = {
+  submissionMonth: IsoMonthStamp;
+  item: UtilisationReportReconciliationSummaryItem;
+};
+
+const mapToSummaryItem = async (bank: Bank, report: UtilisationReport | null): Promise<UtilisationReportReconciliationSummaryItem> => {
+  // TODO FN-???? - Use actual REPORT_NOT_RECEIVED UtilisationReport when added by job. Throw errors when null.
+  if (!report) {
+    return {
+      bank: {
+        id: bank.id,
+        name: bank.name,
+      },
+      status: UTILISATION_REPORT_RECONCILIATION_STATUS.REPORT_NOT_RECEIVED,
+    };
+  }
+
+  const reportData = await getAllUtilisationDataForReport(report);
+
+  // TODO FN-???? - status to be added to report data to allow us to calculate how
+  //  many facilities are left to reconcile
+  const reportedFeesLeftToReconcile = reportData.length;
+
+  return {
+    reportId: report._id,
+    bank: {
+      id: bank.id,
+      name: bank.name,
+    },
+    status: report.status,
+    dateUploaded: report.dateUploaded,
+    totalFeesReported: reportData.length,
+    reportedFeesLeftToReconcile,
+    isPlaceholderReport: report.azureFileInfo === null,
+  };
+};
+
+const mapToSummaryItemForSubmissionMonth = async (
+  bank: Bank,
+  { submissionMonth, report }: UtilisationReportForSubmissionMonth,
+): Promise<SummaryItemForSubmissionMonth> => ({
+  submissionMonth,
+  item: await mapToSummaryItem(bank, report),
+});
+
+const generateMissingReports = ({ from, to }: { from: IsoMonthStamp; to: IsoMonthStamp }): UtilisationReportForSubmissionMonth[] => {
+  const missingSubmissionMonths = eachIsoMonthOfInterval(from, to, {
+    exclusive: true,
+  });
+
+  return missingSubmissionMonths.map((submissionMonth) => ({
+    submissionMonth,
+    report: null,
+  }));
+};
+
+const addNotReceivedReportsAndMapToSubmissionMonth = (
+  reports: UtilisationReport[],
+  currentSubmissionMonth: IsoMonthStamp,
+): UtilisationReportForSubmissionMonth[] => {
+  const currentReportPeriodStart = getReportPeriodStartForSubmissionMonth(currentSubmissionMonth);
+  const previousReportPeriodStart = getPreviousReportPeriodStart(currentReportPeriodStart);
+
+  const reportsOrderedByReportPeriodStartAscending = orderBy(reports, ['year', 'month'], ['asc', 'asc']);
+
+  const updatedReportsWithSubmissionMonth = reportsOrderedByReportPeriodStartAscending.map((report) => {
+    const reportPeriodStart = getReportPeriodStartForUtilisationReport(report);
+    const submissionMonth = getSubmissionMonthForReportPeriodStart(reportPeriodStart);
+    return { submissionMonth, report };
+  });
+
+  // TODO FN-???? - when REPORT_NOT_RECEIVED reports are added by a job we can
```

Comment:
ditto, let's just use FN-1549

Reply:
Have added ref for [FN-1949](https://ukef-dtfs.atlassian.net/browse/FN-1949) (which is for refactoring previous reports)